File: services/Micro Services/job-api/JobApi.Service/Infrastructure/Data/Entities/Job.cs
using Elkhair.Dev.Common.Domain.Entities;
using JobAPI.Contracts.Enums;

namespace JobApi.Infrastructure.Data.Entities;

public class Job : BaseAuditableEntity
{
    public required string Title { get; set; }
    public int CompanyId { get; set; }
    public required string Location { get; set; }
    public required JobType JobType { get; set; }
    public required string AboutRole { get; set; }
    public string? SalaryRange { get; set; }
    public ICollection<Responsibility> Responsibilities { get; set; } = [];
    public ICollection<Qualification> Qualifications { get; set; } = [];
    public Company Company { get; set; } = null!;
}

File: apps/job-admin/src/app/core/types/models/Job.ts
export type Job = {
  uId: string;
  title: string;
  companyUId: string;
  companyName: string;
  location: string;
  jobType: string;
  aboutRole: string;
  salaryRange?: string;
  responsibilities: string[];
  qualifications: string[];
  createdAt: string;
  updatedAt?: string;
}
